custom/CustomCAN.java

```diff
@@ -60,14 +59,10 @@ protected ByteBuffer readBuffer() {
 		idBuffer.put(0, Integer.reverseBytes(messageID));
 		ByteBuffer timestamp = ByteBuffer.allocate(4);
 		ByteBuffer response = null;
-		long start = System.currentTimeMillis();
-		while (System.currentTimeMillis() - start < CustomCAN.CAN_MAX_READ_WAIT) {
-			try {
-				response = CANJNI.FRCNetCommCANSessionMuxReceiveMessage(idBuffer, 0x1fffffff, timestamp);
-				break;
-			}
-			catch (CANMessageNotFoundException e) {}
+		try {
+			response = CANJNI.FRCNetCommCANSessionMuxReceiveMessage(idBuffer, 0x1fffffff, timestamp);
 		}
+		catch (CANMessageNotFoundException e) {}
```

Comment:
I would agree with you completely if I knew when that exception happens and what it means.
(CAN message not found does seem somewhat self-explanatory, but I am concerned that things are not what they seem behind the JNI wall).
It would not hurt to add a warning for it I suppose.